From patchwork Fri Oct 12 17:05:53 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jim Mattson <jmattson@google.com>
X-Patchwork-Id: 10638971
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 3364114E2
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 12 Oct 2018 17:06:00 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 220622BDB3
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 12 Oct 2018 17:06:00 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 161372BF63; Fri, 12 Oct 2018 17:06:00 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-15.5 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,
	USER_IN_DEF_DKIM_WL autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 36A112BDB3
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 12 Oct 2018 17:05:59 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727790AbeJMAjY (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 12 Oct 2018 20:39:24 -0400
Received: from mail-yb1-f202.google.com ([209.85.219.202]:34722 "EHLO
        mail-yb1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726665AbeJMAjX (ORCPT <rfc822;kvm@vger.kernel.org>);
        Fri, 12 Oct 2018 20:39:23 -0400
Received: by mail-yb1-f202.google.com with SMTP id j71-v6so6496616ybg.1
        for <kvm@vger.kernel.org>; Fri, 12 Oct 2018 10:05:57 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:message-id:mime-version:subject:from:to:cc;
        bh=NlXjjJ6xt7q3lqVGxQxqksh8sqSaDWBaIdcAQPhqKJs=;
        b=qiolBLjf0jSyHomQ/OriEINjOuG16nxRWq7GNFtzLZXwnebplvYTO+w6ZAQVuKxgpB
         lKp3Ln2NWgdsp7btiJPW3nSTqF0CDcnwR5GXh6MBW36EOEH9qnDjk1IVvWNpSU8Ujvv9
         vE/vH4RepwBVHWrIlThv2DmtvKoz/tw79JHzjjDLF0ORH+/OgH+LlnihD3OhKEbYc5yM
         AoTDB6Urgm9s1ezBbKizqYMipEZFk5fIvDLcH/z7lDgCCl8ht6ghsOCyFiUe95OA+2q5
         TIG1vRj5XEjj9trWiJUwLqeJnAjE9yOMtR0ZJy7e31d329el5/YvLQMS92jT7xiv2OWg
         G7iQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:message-id:mime-version:subject:from:to:cc;
        bh=NlXjjJ6xt7q3lqVGxQxqksh8sqSaDWBaIdcAQPhqKJs=;
        b=G68Kfao/Skqrghz2e9iCf3yRZhwhmg3uCb8URnRnxvumhkKN3+fw9h03On6fhEXcav
         qPDk/mte5yMoW6sVo6Iyskpuw+8Eti1iSCNkEjr6LjCOokwCvV0F7Fi1QCCD5cqPXOjK
         5t1RHDImW82BpEdeRHTWYPSYaBZX+hWD/DBSMyUl0ivvkG17RzVy1I8Tyu6WI3/lEbPp
         xycmvAmv1Xnb1VhuuU4EUg4RO6spG4c04Ttht4L5zne4Lg4kXln2QxEblmBvTDeKnDag
         9GH+oRhB1wxq4qSvhPnKNIbs0BTx9w0+1O+mofaRiHunzKxg1CKBT2w/y39EcICQV39J
         1pAQ==
X-Gm-Message-State: ABuFfogfRLL1YziI6jzc3a4M49FJ9d0+ZokWdceM0q9Gm849P2hP2HLS
        PwtJecEKadqYurJudPA0FoN7WsFIGCokjE4N/KD9s1GeCdv5z33zpzK1aqZ/FwaGBcH9Q1kDQD/
        vww2W53u/gm4aTxYoZTS44NjjjdLyW5/xl/ejm8pn2/7Rnthybvq1A0IcCxJmhes=
X-Google-Smtp-Source: 
 ACcGV613oyRRV+upBGDCHqnKyEgZX2Mut1B4bdAQiv8sNo7idFvhcCmjgWbixBQCqZQMrDJGaYA9IdRGg5yPPg==
X-Received: by 2002:a25:5d6:: with SMTP id
 205-v6mr4188277ybf.52.1539363956886;
 Fri, 12 Oct 2018 10:05:56 -0700 (PDT)
Date: Fri, 12 Oct 2018 10:05:53 -0700
Message-Id: <20181012170553.209685-1-jmattson@google.com>
Mime-Version: 1.0
X-Mailer: git-send-email 2.19.0.605.g01d371f741-goog
Subject: [PATCH] kvm: x86: #DF only has an error code in protected mode
From: Jim Mattson <jmattson@google.com>
To: kvm@vger.kernel.org
Cc: Jim Mattson <jmattson@google.com>
Content-Type: text/plain; charset="UTF-8"
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Exceptions do not have error codes in real-address mode.

Fixes: 3fd28fce76563 ("KVM: x86: make double/triple fault promotion generic to all exceptions")
Signed-off-by: Jim Mattson <jmattson@google.com>
Reviewed-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 arch/x86/kvm/x86.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index ca717737347e..4ab569171ad1 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -452,7 +452,7 @@ static void kvm_multiple_exception(struct kvm_vcpu *vcpu,
 		 */
 		vcpu->arch.exception.pending = true;
 		vcpu->arch.exception.injected = false;
-		vcpu->arch.exception.has_error_code = true;
+		vcpu->arch.exception.has_error_code = is_protmode(vcpu);
 		vcpu->arch.exception.nr = DF_VECTOR;
 		vcpu->arch.exception.error_code = 0;
 	} else
